### DISCUSSION TITLE
How to get starship prompt to reset/refresh in place?

### DISCUSSION THREAD

**Discussion by manojkarthick** [starship/starship#2036] [General]
I'm currently using a bookmark based jumper of sorts using a Zsh widget that allows you to switch between directories. After switching directories using `cd`, I would like to refresh the prompt to make the prompt display information relevant to the switched directory.

In regular zsh, you can achieve this using a `zle reset-prompt` function to refresh the prompt. Is there any way to do the same with starship (either using a starship command or making starship respect reset-prompt)?

Great project, it's shaved so much off my shell configuration. Thanks a lot for building this project!

---

This is the function I'm using for reference:
```
  # Jump to project bookmark
  jump() {
          local dir
          dir=$(cat ~/code/.bookmarks| fzf ) && cd "$dir"
          zle reset-prompt
}
zle -N jump
```

**Comment by andytom** [comment]
Unfortunately this is a know issue see #353 but there is a work around. If you change your jump function to the following it should work. 
```diff
 # Jump to project bookmark
 jump() {
           local dir
           dir=$(cat ~/code/.bookmarks| fzf ) && cd "$dir"
+          starship_precmd
           zle reset-prompt
 }
 zle -N jump
```

**Comment by manojkarthick** [comment]
Thanks @andytom - the workaround seems to work pretty well!
